pymeasure/instruments/hp/hp437b.py

```diff
@@ -0,0 +1,949 @@
+
+    @limit_low.setter
+    def limit_low(self, limit):
+        """
+        Control the lower limit for the builtin limit checking.
+        """
+        values = [-299.999, 299.999]
+        strict_range(limit, values)
+
+        self.write("LL%7.3fEN" % limit)
+        self.check_errors()
+
+    limit_high_hit = Instrument.measurement(
+        "SM",
+        """
+        Get if the upper limit check got triggered.
+        """,
+        map_values=True,
+        values={True: 1, False: 0},
+        cast=int,
+        get_process=__getstatus(StatusMessage.LimitsStatus),
+    )
+
+    limit_low_hit = Instrument.measurement(
+        "SM",
+        """
+        Get if the lower limit check got triggered.
+        """,
+        map_values=True,
+        values={True: 2, False: 0},
+        cast=int,
+        get_process=__getstatus(StatusMessage.LimitsStatus),
+    )
+
+    power = Instrument.measurement(
+        "",
+        """
+        Measure the power at the power sensor attached to the power meter in the corresponding unit.
+        In case a measurement would be invalid the power meter responds with the value +9.0200e+40.
+        """
+    )
+
+    power_reference_enabled = Instrument.control(
+        "SM", "OC%d",
+        """
+        Control the builtin reference power source 1mW @ 50 MHz.
+        """,
+        map_values=True,
+        values={True: 1, False: 0},
+        cast=int,
+        get_process=__getstatus(StatusMessage.PowerRefStatus),
+        check_set_errors=True
+    )
+
+    offset_enabled = Instrument.control(
+        "SM", "OF%d",
+        """
+        Control the offset being applied.
+        """,
+        map_values=True,
+        values={True: 1, False: 0},
+        cast=int,
+        get_process=__getstatus(StatusMessage.OffsetStatus),
+        check_set_errors=True
+    )
+
+    @property
+    def offset(self):
+        """
+        Control the offset applied to the measured value to compensate for
+        signal gain or loss (for example, to compensate for the loss of a 10 dB
+        directional coupler). Offsets are entered in dB.
+        In case the :attr:`offset_enabled` is false this returns automatically 0.0
+        """
+        if self.offset_enabled:
+            self.write("OS")
+            # returns OFS +00.00 dB
+            display_content = self.display_output
+            assert display_content[0:3] == "OFS"
+            self.write("EX")
+            return float(display_content[4:10])
+        else:
+            return 0.0
+
+    @offset.setter
+    def offset(self, offset):
+        values = [-99.99, 99.99]
+        strict_range(offset, values)
+        self.write("OS%5.2fEN" % offset)
+
+    def reset(self):
+        self.write("*RST")
+
+    def clear_status_registers(self):
+        self.write("*CLS")
+
+    def preset(self):
+        """
+        Sets the power meter to a known state. Preset
+        conditions are shown in the following table.
```

Comment:
known state? Not better a "factory defined state"?

Reply:
That's not exactly the case, some settings remain while others get actually cleared